src/store/DialogStore.js

```diff
@@ -1,15 +1,21 @@
-import { observable, action } from 'mobx';
+import { observable, action, when } from 'mobx';
 import { connect } from './Connect';
 
 let activeDialog;
 
 export default class DialogStore {
     constructor(mainStore) {
         this.mainStore = mainStore;
+        when(() => this.context, () => {
+            this.routingStore.registerDialog(this);
+        });
```

Comment:
@shayanbinary are you still getting `RangeError: Maximum call stack size exceeded` ?

It looks innocent to me.